ui/src/app/views/dataset/component.tsx

```diff
@@ -104,20 +110,23 @@ export const DataSet = withCurrentWorkspace()(class extends React.Component<
   }
 
   receiveDelete() {
-    const {resource, rType} = this.state;
+    const {rType} = this.state;
+    const {namespace, id} = this.props.workspace;
     let call;
-    const id = this.getCurrentResource().id;
-    const {workspaceNamespace, workspaceFirecloudName} = resource;
+    const resourceId = this.getCurrentResource().id;
     if (rType === ResourceType.CONCEPT_SET) {
-      call = conceptSetsApi().deleteConceptSet(workspaceNamespace, workspaceFirecloudName, id);
+      call = conceptSetsApi().deleteConceptSet(namespace, id, resourceId);
     } else {
-      call = cohortsApi().deleteCohort(workspaceNamespace, workspaceFirecloudName, id);
+      call = cohortsApi().deleteCohort(namespace, id, resourceId);
     }
     if (this.state.rType === ResourceType.CONCEPT_SET) {
-      const deleted = this.state.resource.conceptSet;
-      const updatedList = fp.filter(conceptSet => conceptSet.id !== deleted.id,
+      const updatedList = fp.filter(conceptSet => conceptSet.id !== resourceId,
         this.state.conceptSetList);
       this.setState({conceptSetList: updatedList});
+    } else {
+      const updatedList = fp.filter(cohort => cohort.id !== resourceId,
```

Comment:
FYI: formulating this as a `remove` could allow you to take advantage of the shorthand:
```
fp.remove({id: resourceId}, this.state.cohortList)
```